SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +656,237 @@ private void DisplayErrorMessage(string message)
+                    pswHash = name.PasswordHash;
+                    pswSalt = name.Salt;
+                    username = name.Username;
+                    usernameExists = true;
+                }
+            }
+
+            UserData registeredUser = new UserData(username, pswHash, pswSalt, userID);
+            UserData offlineUser = new UserData(UsernameTextBox.Text, null, pswSalt, userID);
+
+            if (UsernameTextBox.Text.Length == 0 || PasswordBox.Password.Length == 0)
+            {
+                LabelEmptyFieldsError.Visibility = Visibility.Visible;
+            }
+            else if (!usernameExists)
+            {
+                LabelUsernameError.Visibility = Visibility.Visible;
+            }
+            else if(registeredUser.PasswordHash != offlineUser.PasswordHash)
+            {
+                LabelPasswordError.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                ShowUserIDLabel.Content = "Your ID: " + userID;
+                ShowUsernameLabel.Content = username;
+
+                GridLogin.Visibility = Visibility.Collapsed; 
+                GridRegistration.Visibility = Visibility.Collapsed;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void RegisterButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            var password = PasswordBoxR.Password;
+            var username = UsernameTextBoxR.Text;
+
+            LabelForUsedUsernameError.Visibility = Visibility.Collapsed;
+            LabelForInvalidUsernameError.Visibility = Visibility.Collapsed;
+            LabelForPasswordError.Visibility = Visibility.Collapsed;
+            LabelForConfirmPasswordError.Visibility = Visibility.Collapsed;
+            LabelForEmptyFieldsError.Visibility = Visibility.Collapsed;
+
+            if (CheckRegisterData(password, username))
+            {
+                userLoginData.AddData(new UserData(username));
+                userLoginData.SaveData();
+                
+                var users = userLoginData.GetAllData();
+                string userID = users[users.Count - 1].ID;
+
+                ShowUserIDLabel.Content = "Your ID: " + userID;
+                ShowUsernameLabel.Content = username;
+
+                GridLogin.Visibility = Visibility.Collapsed;
+                GridRegistration.Visibility = Visibility.Collapsed;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void LoginWhenRegistered_Clicked(object sender, RoutedEventArgs e)
+        {
+            GridLogin.Visibility = Visibility.Visible;
+            GridRegistration.Visibility = Visibility.Collapsed;
+        }
+
+        public bool Duplicates(int randomID)
+        {
+            foreach (var userID in userLoginData.GetAllData())
```

Comment:
The name userID is misleading, since this is a whole userData class, not just the id.